src/Microsoft.AspNet.Mvc.Core/Filters/ProducesAttribute.cs

```diff
@@ -31,7 +32,13 @@ public override void OnResultExecuting([NotNull] ResultExecutingContext context)
 
             if (objectResult != null)
             {
-                SetContentTypes(objectResult.ContentTypes);
+                // Check if FormatFilter has already set the content type
+                // If it has, dont override it
+                var formatFilter = context.Filters.LastOrDefault(f => f is IFormatFilter) as IFormatFilter;
+                if (formatFilter == null || formatFilter.GetContentTypeForCurrentRequest(context) == null)
+                {
+                    SetContentTypes(objectResult.ContentTypes);
```

Comment:
The way this is written this would mean that if I have a IFormatFilter which returns  non null but does not add it to objectResult, it would not add the ones on produces. I think that it should be fine but not sure if this has been discussed. If it has been please add a test case.